.github/workflows/tests_with_database.yml

```diff
@@ -7,6 +7,66 @@ on:
   pull_request: {}
 
 jobs:
+  mysql:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - icinga/icingadb-mysql:5.5
+          - icinga/icingadb-mysql:5.6
+          - mysql:5.7
+          - mysql:latest
+          - icinga/mariadb:10.0
+          - icinga/mariadb:10.1
+          - mariadb:10.2
+          - mariadb:10.3
+          - mariadb:10.4
+          - mariadb:10.5
+          - mariadb:10.6
+          - mariadb:10.7
+          - mariadb:latest
+
+    env:
+      NOTIFICATIONS_TESTS_DB_TYPE: mysql
+      NOTIFICATIONS_TESTS_DB: notifications
+      NOTIFICATIONS_TESTS_DB_USER: root
+      NOTIFICATIONS_TESTS_DB_PASSWORD: notifications
+      NOTIFICATIONS_TESTS_DB_HOST: 127.0.0.1
+      NOTIFICATIONS_TESTS_DB_PORT: 3306
+
+    services:
+      mysql:
+        image: ${{ matrix.image }}
+        env:
+          MYSQL_ROOT_PASSWORD: ${{ env.NOTIFICATIONS_TESTS_DB_PASSWORD }}
+          MYSQL_DATABASE: ${{ env.NOTIFICATIONS_TESTS_DB }}
+        ports:
+          - 3306:3306
+
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: stable
+
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Importing Schema
+        run: >
+          sleep 15;
```

Comment:
Load-bearing sleep? Is there no proper way to wait for the database to be ready?

Reply:
Not in MariaDB 11. Ask Yonas, he'll tell you a horror story about this. TL;DR, IGL now uses sleep, too:

* https://github.com/Icinga/icinga-go-library/pull/21

Reply:
> TL;DR, IGL now uses sleep, too:

That's not true! You've just removed the builtin container health checks from there.

```bash
        options: >-
          --health-cmd ${{ matrix.database.incompat == 'true' && '"healthcheck.sh --connect --innodb_initialized"' || '"mysqladmin ping"' }}
          --health-interval 10s
          --health-timeout 5s
          --health-retries 10
```

Reply:
@Al2Klimov Does the :+1: mean that the sleep can and will be removed?

Reply:
Yes, you shouldn't have to use a plain sleep command, I've just fixed that in IGL too, see https://github.com/Icinga/icinga-go-library/pull/20. Btw, we don't support `mariadb:10.0`, so why did you add it to the matrix? Apart from that, why are you using images from the `icinga/` namespace for mariadb?

Reply:
The vanilla images miss those settings: https://github.com/Icinga/icingadb/blob/6102d9cb2e30a628ceaf79cc4ba01db1ab50aa8f/doc/02-Installation.md#setting-up-a-mysql-or-mariadb-database

Reply:
In IGL the action steps before schema import just turned out to take long enough for MySQL to meanwhile start up.